src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/AmazonSqsTransport.php

```diff
@@ -31,10 +33,17 @@ class AmazonSqsTransport implements TransportInterface, SetupableTransportInterf
     private $receiver;
     private $sender;
 
-    public function __construct(Connection $connection, SerializerInterface $serializer = null)
+    public function __construct(Connection $connection, SerializerInterface $serializer = null, ReceiverInterface $receiver = null, SenderInterface $sender = null)
     {
         $this->connection = $connection;
         $this->serializer = $serializer ?? new PhpSerializer();
+        $this->receiver = $receiver;
+        $this->sender = $sender;
+    }
+
+    public static function create(Connection $connection, ReceiverInterface $receiver, SenderInterface $sender): self
+    {
+        return new self($connection, null, $receiver, $sender);
```

Comment:
What is the benefit over `__construct`?

Reply:
For me it's the clean signature, asking for exactly what is needed, without optionals. But if you want it out, just let me know.

Reply:
Anyway, no reason to slow things down if that's not how you like it. Removed in favour of the main constructor :-)

Reply:
The more you have public method, the harder it is to maintain the project.

In this case, I agreed it provides a cleaner signature, but the benefits is very small compared to passing `null` to the existing constructor.

Thank you for the PR :heart: